pkg/services/resource.go

```diff
@@ -0,0 +1,166 @@
+package services
+
+import (
+	"context"
+
+	"github.com/openshift-online/maestro/pkg/dao"
+	"github.com/openshift-online/maestro/pkg/db"
+	logger "github.com/openshift-online/maestro/pkg/logger"
+
+	"github.com/openshift-online/maestro/pkg/api"
+	"github.com/openshift-online/maestro/pkg/errors"
+)
+
+// This flag will only be used in integration test to prove that the advisory lock works
+var DisableAdvisoryLock = false
+
+type ResourceService interface {
+	Get(ctx context.Context, id string) (*api.Resource, *errors.ServiceError)
+	Create(ctx context.Context, dinosaur *api.Resource) (*api.Resource, *errors.ServiceError)
+	Replace(ctx context.Context, dinosaur *api.Resource) (*api.Resource, *errors.ServiceError)
+	Delete(ctx context.Context, id string) *errors.ServiceError
+	All(ctx context.Context) (api.ResourceList, *errors.ServiceError)
+
+	FindBySpecies(ctx context.Context, species string) (api.ResourceList, *errors.ServiceError)
```

Comment:
Do we need this method? if so, may change it to `FindByTypes`

Reply:
This is template code, we can drop that and add anyone useful, eg, `FindByConsumer` or `FindByCluster`.